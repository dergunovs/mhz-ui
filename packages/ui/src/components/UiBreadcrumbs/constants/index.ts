export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
