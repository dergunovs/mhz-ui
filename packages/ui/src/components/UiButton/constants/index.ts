export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
