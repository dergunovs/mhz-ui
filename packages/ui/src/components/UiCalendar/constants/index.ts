import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
