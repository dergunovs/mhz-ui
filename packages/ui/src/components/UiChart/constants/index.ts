export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
