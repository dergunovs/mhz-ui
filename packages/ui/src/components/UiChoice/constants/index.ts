export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
