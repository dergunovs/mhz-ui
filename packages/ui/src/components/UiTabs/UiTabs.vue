<template>
  <div :class="$style.tabs" data-test="ui-tabs">
    <button
      v-for="tab in props.tabs"
      :key="tab.value"
      @click="emit('update:modelValue', tab.value)"
      :class="$style.tab"
      type="button"
      :data-active="props.modelValue === tab.value"
      data-test="ui-tab"
    >
      {{ tab.title }}
    </button>
  </div>
</template>

<script setup lang="ts">
interface ITab {
  title: string;
  value: string;
}

interface IProps {
  tabs: ITab[];
  modelValue: string;
}

const props = defineProps<IProps>();
const emit = defineEmits<{ 'update:modelValue': [value: string] }>();
</script>

<style module lang="scss">
.tabs {
  display: flex;
  flex-wrap: wrap;
  border-bottom: 2px solid var(--color-primary);
}

.tab {
  padding: 10px 8px;
  font-size: 0.875rem;
  cursor: pointer;
  background-color: var(--color-gray-light);
  border: 0;

  &:hover {
    background-color: var(--color-gray);
  }

  &[data-active='true'] {
    color: var(--color-white);
    background-color: var(--color-primary);
  }
}

@media (max-width: $mobile) {
  .tab {
    padding: 10px 5px;
  }
}
</style>
