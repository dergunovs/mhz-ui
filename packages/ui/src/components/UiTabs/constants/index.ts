export const TABS = [
  { title: 'Занятия', value: 'activities' },
  { title: 'Подходы', value: 'sets' },
  { title: 'Повторы', value: 'repeats' },
  { title: 'Мышцы', value: 'muscles' },
  { title: 'Длительность', value: 'duration' },
];
