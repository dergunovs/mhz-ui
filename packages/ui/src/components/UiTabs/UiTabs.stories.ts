import { Meta, StoryObj } from '@storybook/vue3';
import { html } from 'mhz-helpers';

import { TABS } from './constants';

import { UiTabs } from '@/components';

const meta: Meta<typeof UiTabs> = {
  component: UiTabs,
  args: {
    tabs: TABS,
    modelValue: TABS[0].value,
  },
  argTypes: {},
};

type Story = StoryObj<typeof UiTabs>;

export default meta;

export const Primary: Story = {
  render: (args, { updateArgs }) => ({
    components: { UiTabs },
    setup: () => ({ args, updateArgs }),

    template: html` <UiTabs v-bind="args" @update:modelValue="update" />`,

    methods: {
      update(value: string) {
        updateArgs({ modelValue: value });
      },
    },
  }),
};
