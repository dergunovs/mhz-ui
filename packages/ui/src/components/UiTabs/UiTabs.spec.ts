import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiTabs from './UiTabs.vue';
import { TABS } from './constants';

import { wrapperFactory } from '@/test';

const tabs = dataTest('ui-tabs');
const tab = dataTest('ui-tab');

let wrapper: VueWrapper<InstanceType<typeof UiTabs>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiTabs, { tabs: TABS, modelValue: TABS[0].value });
});

enableAutoUnmount(afterEach);

describe('UiTabs', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiTabs)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders tabs correctly', async () => {
    expect(wrapper.findAll(tab).length).toBe(TABS.length);

    expect(wrapper.find(tab).text()).toBe(TABS[0].title);
    expect(wrapper.find(tab).attributes('data-active')).toBe('true');
  });

  it('updates modelValue when tab is clicked', async () => {
    const secondTab = wrapper.findAll(tab)[1];

    await secondTab.trigger('click');

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([TABS[1].value]);
  });

  it('sets active tab correctly', async () => {
    await wrapper.setProps({ modelValue: TABS[1].value });

    const tabElements = wrapper.findAll(tab);

    expect(tabElements[0].attributes('data-active')).toBe('false');
    expect(tabElements[1].attributes('data-active')).toBe('true');
    expect(tabElements[2].attributes('data-active')).toBe('false');
  });

  it('handles empty tabs array', async () => {
    await wrapper.setProps({ tabs: [] });

    expect(wrapper.findAll(tab).length).toBe(0);
    expect(wrapper.find(tabs).exists()).toBe(true);
  });
});
