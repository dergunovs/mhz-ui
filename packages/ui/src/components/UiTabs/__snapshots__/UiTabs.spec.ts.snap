// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiTabs > matches snapshot 1`] = `"<div class="_tabs_24df60" data-test="ui-tabs"><button class="_tab_24df60" type="button" data-active="true" data-test="ui-tab">Занятия</button><button class="_tab_24df60" type="button" data-active="false" data-test="ui-tab">Подходы</button><button class="_tab_24df60" type="button" data-active="false" data-test="ui-tab">Повторы</button><button class="_tab_24df60" type="button" data-active="false" data-test="ui-tab">Мышцы</button><button class="_tab_24df60" type="button" data-active="false" data-test="ui-tab">Время</button></div>"`;
